frontend/lib/api/openapi.test.ts

```diff
@@ -0,0 +1,22 @@
+import fs from "fs";
+import path from "path";
+import { describe, expect, test } from "vitest";
+
+import { generate } from "../../scripts/openapi/generator";
+
+const FRONTEND_ROOT = path.join(__dirname, "../..");
+const BACKEND_ROOT = path.join(FRONTEND_ROOT, "..", "backend");
+
+const OPEN_API_JSON_PATH = `${BACKEND_ROOT}/openapi.json`;
+const CURRENT_SPEC_PATH = `${FRONTEND_ROOT}/lib/api/gen/openapi.ts`;
+
+describe("openapi spec", () => {
+  test("openapi spec is up to date", async () => {
+    const openAPIJSONString = fs.readFileSync(OPEN_API_JSON_PATH, "utf8");
+
+    const latestSpecString = await generate(openAPIJSONString);
+    const currentSpecString = fs.readFileSync(CURRENT_SPEC_PATH, "utf8");
```

Comment:
You could use the async `fs.readFile` function instead.

Reply:
my typescript thinks its not promisified yet.

Reply:
updated, but it's a bit iffy using a different import. works here but makes the gen_openapi_types file more complex

Reply:
Let's keep it to only changing the test then!